src/vs/workbench/services/activity/common/activityService.ts

```diff
@@ -70,4 +70,24 @@ export interface IActivityService {
 	 * Clears activity shown in the activitybar for the given viewlet or panel.
 	 */
 	clearActivity(compositeId: string): void;
-}
\ No newline at end of file
+
+	/**
+	 * Get registered external viewlets' info for populating 'Toggle Custom Explorer' command picks.
+	 */
+	getInfoForRegisteredViewlets(): {
+		[viewletId: string]: {
+			isEnabled: boolean;
+			treeLabel: string;
+		}
+	};
+
+	/**
+	 * Enable/disable an external viewlet.
+	 */
+	toggleViewlet(viewletId: string): void;
+
+	/**
+	 * Get the external viewlet id that is about to open.
+	 */
+	getExternalViewletIdToOpen(): string;
+}
```

Comment:
[needs thinking] @bpasero will know better, but I have a strong feeling that the `IActivityService` isn't the right place for viewlet specifics. Isn't there an `IViewletService` and wouldn't that be the better place for this?